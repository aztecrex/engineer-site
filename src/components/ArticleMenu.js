import React from 'react';
import {Link} from 'react-router';
import R from 'ramda';

class ArticleMenu extends React.Component {

  keyedMenuItem(entry) {
    return (
      <li className="article-menu-item" key={entry.id}>
        <Link to={"article/" + entry.id}>
          {entry.title}
        </Link>
      </li>
    );
  }

  render() {
    const directory = this.props.directory || [];
    return (
      <ul className="article-menu">
        {R.map(this.keyedMenuItem, directory)}
      </ul>
    );
  }
}

export default ArticleMenu;
