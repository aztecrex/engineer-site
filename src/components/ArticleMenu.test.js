import React from 'react';
import ArticleMenu from './ArticleMenu';
import ezJson from 'enzyme-to-json';



describe ('render', () => {

  it('renders entries', () => {
    // given
    const directory = [
      {
        "id": "one",
        "title": "One"
      },
      {
        "id": "two",
        "title": "Two"
      }
    ];

    // when
    const rendered = shallow(<ArticleMenu directory={directory} />);

    // then
    expect(ezJson(rendered)).toMatchSnapshot();

  });

  it('renders empty entries', () => {
    // when
    const rendered = shallow(<ArticleMenu directory={[]} />);

    // then
    expect(ezJson(rendered)).toMatchSnapshot();

  });

  it('renders missing entries', () => {
    // when
    const rendered = shallow(<ArticleMenu />);

    // then
    expect(ezJson(rendered)).toMatchSnapshot();

  });


});

// describe('rendering', () => {
//   it('renders an article as a link', () => {
//     // given
//     let id = "article-a";
//     let title = "Article Title";
//     const directory = [
//       {
//         "id": id,
//         "title": title
//       }
//     ];
//     let state = makeState(directory);
//     let embedded = embed(ArticleMenu, state);
//     // when
//     let wrapper = mount(embedded);
//
//     // then
//     let links = wrapper.find(Link);
//     expect(links.length).toBe(1);
//     expect(links.first().text()).toEqual(title);
//     expect(links.prop('to')).toEqual('article/' + id);
//
//   });
//
//   it('renders multiple articles', () => {
//     // given
//     const directory = [
//       {
//         "id": "one",
//         "title": "One"
//       },
//       {
//         "id": "two",
//         "title": "Two"
//       }
//     ];
//     let state = makeState(directory);
//     let embedded = embed(ArticleMenu, state);
//
//     // when
//     let wrapper = mount(embedded);
//
//     // then
//     let links = wrapper.find(Link);
//     expect(links.length).toBe(2);
//
//   });
//
//   it('renders empty list if directory missing', () => {
//     // given
//     let state = {articles:{}};
//     let embedded = embed(ArticleMenu, state);
//
//     // when
//     let wrapper = render(embedded);
//
//     // then
//     let ul = wrapper.find('ul');
//     expect(ul.length).toEqual(1);
//
//     let as = ul.find('li a');
//     expect(as.length).toEqual(0);
//   });
//
//   it('renders empty list if articles missing', () => {
//     // given
//     let state = {};
//     let embedded = embed(ArticleMenu, state);
//
//     // when
//     let wrapper = render(embedded);
//
//     // then
//     let ul = wrapper.find('ul');
//     expect(ul.length).toEqual(1);
//
//     let as = ul.find('li a');
//     expect(as.length).toEqual(0);
//   });
// });
