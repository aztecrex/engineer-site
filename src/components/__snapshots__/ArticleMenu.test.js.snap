exports[`render renders empty entries 1`] = `
<ul
  className="article-menu" />
`;

exports[`render renders entries 1`] = `
<ul
  className="article-menu">
  <li
    className="article-menu-item">
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="article/one">
      One
    </Link>
  </li>
  <li
    className="article-menu-item">
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="article/two">
      Two
    </Link>
  </li>
</ul>
`;

exports[`render renders missing entries 1`] = `
<ul
  className="article-menu" />
`;
